src/resources/resources.controller.ts

```diff
@@ -0,0 +1,74 @@
+import {
+    Controller,
+    Get,
+    Post,
+    Body,
+    Patch,
+    Param,
+    Delete,
+    ParseIntPipe,
+} from "@nestjs/common";
+import { ApiOperation, ApiTags } from "@nestjs/swagger";
+import { ResourcesService } from "./resources.service";
+import { CreateResourceDto } from "./dto/create-resource.dto";
+import { UpdateResourceDto } from "./dto/update-resource.dto";
+import { DeleteResourceDto } from "./dto/delete-resource.dto";
+
+@Controller("team-resources")
+@ApiTags("team resources")
+export class ResourcesController {
+    constructor(private readonly resourcesService: ResourcesService) {}
+
+    @ApiOperation({
+        description:
+            "Adds a URL with title to the team's resources, addedBy: teamMemberId (int)",
+    })
+    @Post(":teamMemberId")
+    createNewResource(
+        @Param("teamMemberId", ParseIntPipe) teamMemberId: number,
+        @Body() createResourceDto: CreateResourceDto,
+    ) {
+        return this.resourcesService.createNewResource(
+            createResourceDto,
+            teamMemberId,
+        );
+    }
+
+    @ApiOperation({
+        description: "Gets all resources added by a team given a teamId (int)",
+    })
+    @Get(":teamId")
+    findAllResources(@Param("teamId", ParseIntPipe) teamId: number) {
+        return this.resourcesService.findAllResources(teamId);
+    }
+
+    @ApiOperation({
+        description:
+            "Edit URL/title for a resource if teamMemberId (int) matches logged in user",
+    })
+    @Patch(":teamMemberId/:resourceId")
```

Comment:
this endpoint probably doesn't follow the naming convention we have been following
https://restfulapi.net/resource-naming/

Reply:
I don't see which convention is being broken -- seems like the kebab case convention doesn't apply to param names?

Reply:
you can read the linked article for example, and the rest of the project we have been using this, and Kenny has fixed some older ones in the recent PR #28 

it would be something like `team-members/:teamMemberId/resources/:resourceId` in this case. but as discussed we should be passing teamId instead of teamMemberId here

Reply:
Updated the code following the way ideations did it:
/api/v1/teams/{teamId}/team-resources    for POST/GET
/api/v1/team-resources/{resourceId}    for PATCH/DELETE

Wasn't sure if perhaps PATCH/DELETE should be /api/v1/teams/{teamId}/team-resources/{resourceId} instead, ideations did it the like the first way and tech stacks the second way.